pkg/apis/keycloak/v1alpha1/keycloakclient_types.go

```diff
@@ -149,6 +155,147 @@ type KeycloakProtocolMapper struct {
 	Config map[string]string `json:"config,omitempty"`
 }
 
+type KeycloakResourceServer struct {
+	// True if resources should be managed remotely by the resource server.
+	// +optional
+	AllowRemoteResourceManagement bool `json:"allowRemoteResourceManagement,omitempty"`
+	// Client ID.
+	// +optional
+	ClientID string `json:"clientId,omitempty"`
+	// The decision strategy dictates how permissions are evaluated and how a
+	// final decision is obtained. 'Affirmative' means that at least one
+	// permission must evaluate to a positive decision in order to grant access
+	// to a resource and its scopes. 'Unanimous' means that all permissions must
+	// evaluate to a positive decision in order for the final decision to be also positive.
+	// +optional
+	DecisionStrategy string `json:"decisionStrategy,omitempty"`
+	// ID.
+	// +optional
+	ID string `json:"id,omitempty"`
+	// Name.
+	// +optional
+	Name string `json:"name,omitempty"`
+	// Policies.
+	// +optional
+	Policies []KeycloakPolicy `json:"policies,omitempty"`
+	// The policy enforcement mode dictates how policies are enforced when evaluating authorization requests.
+	// 'Enforcing' means requests are denied by default even when there is no policy associated with a given resource.
+	// 'Permissive' means requests are allowed even when there is no policy associated with a given resource.
+	// 'Disabled' completely disables the evaluation of policies and allows access to any resource.
+	// +optional
+	PolicyEnforcementMode string `json:"policyEnforcementMode,omitempty"`
+	// Resources.
+	// +optional
+	Resources []KeycloakResource `json:"resources,omitempty"`
+	// Authorization Scopes.
+	// +optional
+	Scopes []KeycloakScope `json:"scopes,omitempty"`
+}
+
+type KeycloakPolicy struct {
+	// Config.
+	// +optional
+	Config map[string]string `json:"config,omitempty"`
+	// The decision strategy dictates how the policies associated with a given permission are evaluated and how
+	// a final decision is obtained. 'Affirmative' means that at least one policy must evaluate to a positive
+	// decision in order for the final decision to be also positive. 'Unanimous' means that all policies must
+	// evaluate to a positive decision in order for the final decision to be also positive. 'Consensus' means
+	// that the number of positive decisions must be greater than the number of negative decisions. If the number
+	// of positive and negative is the same, the final decision will be negative.
+	// +optional
+	DecisionStrategy string `json:"decisionStrategy,omitempty"`
```

Comment:
Opted for `string` here to push validation of enum values within Keycloak Admin REST API.

Reply:
Sounds good.